pints/_mcmc/_result.py

```diff
@@ -0,0 +1,130 @@
+#
+# MCMC result object
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+# Some code in this file was adapted from Myokit (see http://myokit.org)
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints._diagnostics as diagnostics
+import numpy as np
+from tabulate import tabulate
+
+
+class McmcResultObject(object):
+    """
+    Class that represents the result of running MCMC and
+    calculates key parameter summaries which are printed to the
+    console, including the posterior mean, standard deviation,
+    quantiles, rhat and effective sample size
+    """
+
+    def __init__(self, chains):
+        self._stacked = np.vstack(chains)
+        self._mean = np.mean(self._stacked, axis=0)
+        self._std = np.std(self._stacked, axis=0)
+        self._quantiles = np.percentile(self._stacked, [2.5, 25, 50,
+                                                        75, 97.5], axis=0)
+        self._ess = diagnostics.effective_sample_size(self._stacked)
+        self._num_params = self._stacked.shape[1]
+        self._num_chains = len(chains)
+
+        # If there is more than 1 chain calculate rhat
+        # otherwise return NA
+        if self._num_chains > 1:
+            self._rhat = diagnostics.rhat_all_params(chains)
+        else:
+            self._rhat = np.repeat("NA", self._num_params)
+
+        self._summary_list = None
+        self.make_summary()
+
```

Comment:
Hi Ben,

Maybe we should also add function to return the chain itself (as a numpy array object)! I think it will be especially useful if later we return this class object in the MCMC `run()`.

Best,
Chon